### PR TITLE
feat(controller): Add support for Docker workflow executor for Windows nodes

### PR DESCRIPTION
**Pull request by lippertmarkus**
Argoexec already works with Windows Containers but didn't support `docker` workflow executor yet as Windows Containers require a different way of mounting the Docker socket (see https://github.com/kubernetes/kubernetes/pull/69484):
- Path: `\\.\pipe\docker_engine`
- Type: `null`
- ReadOnly: `false`

I adapted the workflow controller to allow for different volume setups for Windows or Linux Steps of a workflow. When the following `nodeSelector` entry is present for a step in a workflow, then the volume setup for Windows is used, else the setup for Linux is used:
```yaml
nodeSelector:
  kubernetes.io/os: windows
```

I also adapted the documentation as you not longer need to use the `kubelet` workflow executor.

Checklist:
* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed the CLA.
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
  - not needed
* [x] My builds are green. Try syncing with master if they are not. 
* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

### CODE REVIEW

File: workflow/controller/workflowpod.go [hunk 1]
```diff
@@ -53,16 +53,48 @@ var (
...
 }
 
-func (woc *wfOperationCtx) getVolumeDockerSock() apiv1.Volume {
-	dockerSockPath := "/var/run/docker.sock"
+func getDockerSockReadOnly(tmpl *wfv1.Template) bool {
+	return !hasWindowsOsNodeSelector(tmpl.NodeSelector)
+}
+
+func getDockerSockPath(tmpl *wfv1.Template) string {
+	if hasWindowsOsNodeSelector(tmpl.NodeSelector) {
```
Comment (alexec) [inline workflow/controller/workflowpod.go:69]: tests?

Comment (lippertmarkus) [reply workflow/controller/workflowpod.go:69]: added

File: workflow/controller/workflowpod.go [hunk 2]
```diff
@@ -53,16 +53,48 @@ var (
...
+
+func getVolumeHostPathType(tmpl *wfv1.Template) *apiv1.HostPathType {
+	if hasWindowsOsNodeSelector(tmpl.NodeSelector) {
+		return nil
+	}
+
+	return &hostPathSocket
+}
+
+func hasWindowsOsNodeSelector(nodeSelector map[string]string) bool {
```
Comment (alexec) [inline workflow/controller/workflowpod.go:84]: should be `hasWindowsOSNodeSelector` by Golang convention

Comment (lippertmarkus) [reply workflow/controller/workflowpod.go]: done

**Comment by lippertmarkus** [comment]
Thanks for reviewing, I fixed all conflicts now.

**Pull request merged**